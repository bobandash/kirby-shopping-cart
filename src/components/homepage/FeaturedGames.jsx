import styles from './FeaturedGames.module.css';
import sharedStyles from './shared.module.css';
import kirbyForgottenLand from '../../assets/kriby-forgotten-land-promo.jpg'

function FeaturedGames(){
  return (
    <section className = {styles["featured-items-container"]}>
      <div className = {sharedStyles.container}>
        <h1 className = {sharedStyles.header}>Featured Games</h1>
        <div className = {styles["game-images-container"]}>
          <button className = {styles["caret-btn"]}><i className="fa-solid fa-angle-left"></i></button>
          <img src = {kirbyForgottenLand} className = {styles["game-image"]} />
          <button className = {styles["caret-btn"]}><i className="fa-solid fa-angle-right"></i></button>
        </div>
        
      </div>
    </section>
  )
}

export default FeaturedGames;