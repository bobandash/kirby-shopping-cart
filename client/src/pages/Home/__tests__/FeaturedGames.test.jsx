import FeaturedGames from "../FeaturedGames";
import { render, screen } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import { act } from "react-dom/test-utils";

describe("Featured Games Component", () => {
  const mockGames = [
    {
      id: 1,
      image: "test1",
      price: 1,
      title: "test1"
    },
    {
      id: 2,
      image: "test2",
      price: 2,
      title: "test2"
    },
    {
      id: 3,
      image: "test3",
      price: 3,
      title: "test3"
    }
  ];
  describe("DOM", () => {
    it("renders featured games component", () => {
      const { container } = render(
        <BrowserRouter>
          <FeaturedGames games={mockGames} />
        </BrowserRouter>
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("product carousel", () => {
    it("click right arrow button, shows next game", async () => {
      render(
        <BrowserRouter>
          <FeaturedGames games={mockGames} />
        </BrowserRouter>
      );
      const user = userEvent.setup();
      const rightArrowBtn = screen.getByRole("button", { name: /increment active game number/i });
      await user.click(rightArrowBtn);
      expect(screen.getByAltText("test2")).toBeInTheDocument();
    });

    it("click prev arrow button, shows prev game", async () => {
      render(
        <BrowserRouter>
          <FeaturedGames games={mockGames} />
        </BrowserRouter>
      );
      const user = userEvent.setup();
      const leftArrowBtn = screen.getByRole("button", { name: /decrement active game number/i });
      await user.click(leftArrowBtn);
      expect(screen.getByAltText("test3")).toBeInTheDocument();
    });

    it("click bottom button, navigate to game in the index", async () => {
      render(
        <BrowserRouter>
          <FeaturedGames games={mockGames} />
        </BrowserRouter>
      );
      const user = userEvent.setup();
      const allCircleNavBtns = screen.getAllByRole("button", { name: /circle-nav/i });
      await user.click(allCircleNavBtns[1]);
      expect(screen.getByAltText("test2")).toBeInTheDocument();
      await user.click(allCircleNavBtns[0]);
      expect(screen.getByAltText("test1")).toBeInTheDocument();
    });

    it("click game, navigate to product id page", async () => {
      render(
        <BrowserRouter>
          <FeaturedGames games={mockGames} />
        </BrowserRouter>
      );
      const user = userEvent.setup();
      const firstGame = screen.getByAltText(mockGames[0].title);
      await user.click(firstGame);
      expect(window.location.pathname).toBe("/products/" + mockGames[0].id);
    });
  });

  it("games automatically move to the next game after ", async () => {
    vi.useFakeTimers();
    render(
      <BrowserRouter>
        <FeaturedGames games={mockGames} />
      </BrowserRouter>
    );
    act(() => {
      vi.advanceTimersByTime(11000);
    });
    expect(screen.getByAltText(mockGames[1].title)).toBeInTheDocument();
  });
});
