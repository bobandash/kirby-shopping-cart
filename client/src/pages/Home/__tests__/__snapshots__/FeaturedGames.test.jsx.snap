// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Featured Games Component > DOM > renders featured games component 1`] = `
<div>
  <section
    class="_featured-items-container_4c6bb5 _divider_a8154b"
  >
    <div
      class="_container_8b8d20"
    >
      <h1
        class="_header_8b8d20"
      >
        Featured Games
      </h1>
      <div
        class="_game-images-container_4c6bb5"
      >
        <button
          class="_caret-btn_4c6bb5"
        >
          <i
            aria-label="Decrement Active Game Number"
            class="fa-solid fa-angle-left"
          />
        </button>
        <div
          class="_clickable-images-container_4c6bb5"
        >
          <img
            alt="test1"
            class="_game-image_4c6bb5"
            src="test1"
          />
        </div>
        <button
          class="_caret-btn_4c6bb5"
        >
          <i
            aria-label="Increment Active Game Number"
            class="fa-solid fa-angle-right"
          />
        </button>
      </div>
      <div
        class="_circle-nav-container_4c6bb5"
      >
        <button
          aria-label="circle-nav"
          class="_active_4c6bb5 _circle-nav_4c6bb5"
        />
        <button
          aria-label="circle-nav"
          class="_circle-nav_4c6bb5"
        />
        <button
          aria-label="circle-nav"
          class="_circle-nav_4c6bb5"
        />
      </div>
    </div>
  </section>
</div>
`;
